### PR TITLE
Move sign method from Client to Auth

### PR DESCRIPTION
**Pull request by rpbaptist**
`Auth` is still called in `Client`, but only once with two arguments

### CODE REVIEW

**Comment by tomkr** [comment]
@rpbaptist Some things seem to disappear that are probably valuable too keep around: specs and comments at least. Would you mind looking into this?

**Comment by rpbaptist** [comment]
The specs are removed because some of the methods which were exposed are now internals.

I agree we no longer test _how_ things are done, but rather if the response is as expected according to the message.

I will check the comments to see what was lost and what will be helpful.

**Comment by rpbaptist** [comment]
Some things were also tested on different levels and now in one place. I don't think we actually lost any coverage here. Please let me know if you believe otherwise.

File: lib/poms/api/drivers/net_http.rb [hunk 1]
```diff
@@ -60,8 +60,7 @@ def request_to_net_http_request(request_description)
           elsif request_description.post?
             Net::HTTP::Post.new(uri)
           else
-            raise ArgumentError,
-                  'can only execute GET or POST requests'
+            raise ArgumentError, 'can only execute GET or POST requests'
```
Comment (rpbaptist) [inline lib/poms/api/drivers/net_http.rb]: Eerlijk gezegd heb ik deze per ongeluk gecommit.

**Pull request merged**